### PR TITLE
Add token selection scripts

### PR DESCRIPTION
**Pull request by amyxlu**
## Description
Scripts for token selection by assessing per-token loss under a pretrained autoregressive model (ProtGPT2):
* Distributed dataloading by pre-specifying the NumPy offsets in the FASTA loader
* Load ProtGPT2 model as option for all Lbster users
* Launch SLURM jobs for large scale inference and saving per-token data to Parquet file

## Type of Change
- [ ] Bug fix
- [x] New feature  
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

## Testing
- [ ] Tests pass locally
- [ ] Added new tests for new functionality
- [x] Updated existing tests if needed

## Checklist
- [x] Code follows style guidelines
- [x] Self-review completed
- [x] Documentation updated if needed
- [x] No breaking changes (or clearly documented)

### CODE REVIEW

File: token_selection/scripts/save_token_losses.slrm [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (amyxlu) [reply token_selection/scripts/save_token_losses.slrm]: Thanks for the note -- moved this in the latest update!

File: token_selection/scripts/save_token_losses.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+torch.set_float32_matmul_precision('high')
+
+def get_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--fasta_file",
+        type=str,
+        default="/data/lux70/data/uniref90/partial.fasta",
```
[the remark this answers is not in the document]
Comment (amyxlu) [reply token_selection/scripts/save_token_losses.py]: Thanks -- defaults are removed and help prompts added in the latest update.

File: token_selection/scripts/inference.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
+import os
+import glob
+import numpy as np
+import pandas as pd
+import torch
+import torch.distributed as dist
+from torch.utils.data import Dataset, DataLoader
+from torch.utils.data.distributed import DistributedSampler
+
+class ShardedParquetDataset(Dataset):
```
[the remark this answers is not in the document]
Comment (amyxlu) [reply token_selection/scripts/inference.py]: This has been moved to the `datasets` module in the latest update.

File: token_selection/scripts/save_token_losses.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+from torch.nn.parallel import DistributedDataParallel
+from torch.nn.functional import cross_entropy
+
+from lobster.datasets import FASTADataset
+
+
+torch.set_float32_matmul_precision("high")
+
+
+def get_args():
```
Comment (karinazad) [inline token_selection/scripts/save_token_losses.py:23]: can we use hydra for this?

File: token_selection/scripts/save_token_losses.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    cur_num_in_shard = 0
+
+    for batch in dataloader:
+        with torch.no_grad():
+            outputs = compute_loss(batch, ddp_model, tokenizer, args.max_length, device)
+            results_tmp_list.extend(outputs)
+            cur_num_in_shard += len(outputs)
+
+            if cur_num_in_shard >= args.max_num_per_shard:
+                print(f"Saving shard {cur_shard_num} to {output_file}...")
```
Comment (karinazad) [inline token_selection/scripts/save_token_losses.py:192]: output_file seems to be defined only in the next line - does this work?

File: token_selection/scripts/save_token_losses.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+                output_file = output_dir / f"rank_{rank:02}_shard_{cur_shard_num:06}.parquet"
+                pd.DataFrame(results_tmp_list).to_parquet(output_file, engine="pyarrow", index=False)
+
+                cur_shard_num += 1
+                cur_num_in_shard = 0
+                results_tmp_list = []
```
Comment (karinazad) [inline token_selection/scripts/save_token_losses.py:198]: I think leftover results are not saved? if results_tmp_list has data when loop ends, it might be lost

File: src/lobster/datasets/_sharded_parquet_dataset.py [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+            threshold = threshold_tensor.item()
+
+        print(f"Using loss threshold: {threshold}")
+        return threshold
+
+    def _load_sequence_metadata(self):
+        """Load sequence metadata from assigned shards."""
+        sequences = []
+
+        for shard_file in self.shard_files:
```
Comment (karinazad) [inline src/lobster/datasets/_sharded_parquet_dataset.py:95]: why not just load all shards at once with `pd.read_parquet`?

**Comment by karinazad** [comment]
could you pls add tests?
